Lib/test/datetimetester.py

```diff
@@ -1106,6 +1106,65 @@ def test_delta_non_days_ignored(self):
         dt2 = dt - delta
         self.assertEqual(dt2, dt - days)
 
+    def test_strptime(self):
+        string = '2004-12-01'
+        format = '%Y-%m-%d'
+        expected = _strptime._strptime_datetime_date(date, string, format)
+        got = date.strptime(string, format)
+        self.assertEqual(expected, got)
+        self.assertIs(type(expected), date)
+        self.assertIs(type(got), date)
+
+        # bpo-34482: Check that surrogates are handled properly.
+        inputs = [
+            ('2004-12\ud80001', '%Y-%m\ud800%d'),
+            ('2004\ud80012-01', '%Y\ud800%m-%d'),
+        ]
+        for string, format in inputs:
+            with self.subTest(string=string, format=format):
+                expected = _strptime._strptime_datetime_date(date, string,
+                                                             format)
+                got = date.strptime(string, format)
+                self.assertEqual(expected, got)
```

Comment:
I think it would be better to test that this gives the right result rather than test that it does the same thing as `_strptime._stptime_datetime_date`.

I also think it would be best to fold these all into one parameterized test. I am also thinking we should add more test cases (a lot of these are covered by other tests to the underlying implementation, but in case those implementation details change, it's good to test the public-facing portion of this):

```suggestion
        inputs = [
            # Basic valid cases
            (date(1998, 2, 3), '1998-02-03', '%Y-%m-%d'),
            (date(2004, 12, 2), '2004-12-02', '%Y-%m-%d'),

            # Edge cases: Leap year
            (date(2020, 2, 29), '2020-02-29', '%Y-%m-%d'),  # Valid leap year date

            # bpo-34482: Handle surrogate pairs
            (date(2004, 12, 2), '2004-12\ud80002', '%Y-%m\ud800%d'),
            (date(2004, 12, 2), '2004\ud80012-02', '%Y\ud800%m-%d'),

            # Month/day variations
            (date(2004, 2, 1), '2004-02', '%Y-%m'),  # No day provided
            (date(2004, 2, 1), '02-2004', '%m-%Y'),  # Month and year swapped

            # Different day-month-year formats
            (date(2004, 12, 2), '01/12/2004', '%d/%m/%Y'),  # Day/Month/Year
            (date(2004, 12, 2), '12/01/2004', '%m/%d/%Y'),  # Month/Day/Year

            # Different separators
            (date(2023, 9, 24), '24.09.2023', '%d.%m.%Y'),  # Dots as separators
            (date(2023, 9, 24), '24-09-2023', '%d-%m-%Y'),  # Dashes
            (date(2023, 9, 24), '2023/09/24', '%Y/%m/%d'),  # Slashes

            # Handling years with fewer digits
            (date(127, 2, 3), '0127-02-03', '%Y-%m-%d'),
            (date(99, 2, 3), '0099-02-03', '%Y-%m-%d'),
            (date(5, 2, 3), '0005-02-03', '%Y-%m-%d'),

            # Variations on ISO 8601 format
            (date(2023, 9, 24), '2023-W39-1', '%G-W%V-%u'),  # ISO week date (Week 39, Monday)
            (date(2023, 9, 24), '2023-267', '%Y-%j'),  # Year and day of the year (Julian)
        ]
        for expected, string, format in inputs:
            with self.subTest(string=string, format=format):
                got = date.strptime(string, format)
                self.assertEqual(expected, got)
                self.assertIs(type(got), date)
```